### DISCUSSION TITLE
How to get the geo coordinates of a pixel within a bounding box?

### DISCUSSION THREAD

**Discussion by rsousa95** [torchgeo/torchgeo#1032] [Q&A] [answered]
Sorry if this is a beginner question, but I'm not finding the answer anywhere. 

I created a RasterDataset and I query it the section I want via a BoundingBox. It returns me a dictionary containing:
`image`, a torch tensor with the image; 
`crs`, the coordinate reference system;
`bbox`, the bounding box information regarding the section in question.

Now, I need to get the geographic coordinates of a pixel point within this image. Let's say I want the central point. I'd want something like 
`center_x, center_y = get_coordinates_of_pixel(x=image.width//2, y=image.height//2, crs, bbox)`. 
How can I do this?

**Comment by adamjstewart** [comment, marked as answer]
You can get this information from the bounding box:
```python
bbox = sample["bbox"]
center_x = (bbox.minx + bbox.maxx) / 2
center_y = (bbox.miny + bbox.maxy) / 2
```

**Reply by calebrob6** [reply]
The offsets for each pixel can be calculated in terms of the RasterDataset's `res` property, i.e. you should be able to do something like `bbox.miny + (y * ds.res)`.

**Reply by adamjstewart** [reply]
Ah yes, my answer is for geographic coordinates, @calebrob6's answer is for pixel coordinates. Wasn't sure which you were asking about.

**Reply by adamjstewart** [reply]
Nvm, both of these are for geographic coords and should give the same answer.

**Reply by rsousa95** [reply]
Both of these got me what I wanted, thank you! @calebrob6's answer is the more generic one for any point but both work.

Just to add something in case someone stumbles upon this in the future, I did have to do something like:
`geo_x = bbox.minx + (pixel_x * dataset.res)`
`geo_y = bbox.maxy - (pixel_y * dataset.res)`

This is because the reference systems are different. From what I understand, image tensors have their origin in the upper left corner, x axis to the right, y axis downwards. The tiff I was working on has the origin in the lower left corner, x axis to the right, y axis upwards. Hence this translation and inversion in the y axis.
